common/network_manager_utils.go

```diff
@@ -0,0 +1,76 @@
+package common
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	k6lib "go.k6.io/k6/lib"
+	k6netext "go.k6.io/k6/lib/netext"
+	k6types "go.k6.io/k6/lib/types"
+)
+
+func checkBlockedHosts(host string, blockedHosts *k6types.HostnameTrie) error {
+	if blockedHosts == nil {
+		return nil
+	}
+	if match, blocked := blockedHosts.Contains(host); blocked {
+		return fmt.Errorf("hostname %s is in a blocked pattern (%s)", host, match)
```

Comment:
```suggestion
		return fmt.Errorf("hostname %q is in a blocked pattern %q", host, match)
```

Reply:
I don't see a reason to use both parenthesis and quotes... The parenthesis make sense for the blocked pattern, since it's additional information, but not for the hostname, since the message is "hostname X...".

So I'll change it to `hostname %q is in a blocked pattern (%s)`.

Though I kind of think quotes are not needed even for hostname, since it usually won't include whitespace. `hostname "test.k6.io" ...` looks weird IMO.

So I'd rather leave it as is? :sweat_smile:

Reply:
So:

```suggestion
		return fmt.Errorf("hostname %s is in a blocked pattern %q", host, match)
```

Then again, never trust what you're going to get in a string.

Reply:
> never trust what you're going to get in a string

This doesn't really apply here. There's no risk that quoting the string here would prevent :man_shrugging:

It's a style preference (which I thought we weren't going to bring up in PRs?), and TBH I prefer the previous parenthesis usage, but we've bikeshed about this enough, so it's changed in b8eef0e.